apps/browser/qml/pages/SettingsPage.qml

```diff
@@ -67,22 +67,26 @@ Page {
                 menu: ContextMenu {
                     Repeater {
                         model: SearchEngineModel
-                        delegate: Component {
-                            MenuItem {
-                                text: title
-                                Component.onCompleted: {
-                                    if (text && (text === searchEngineConfig.value)) {
-                                        searchEngine.currentIndex = index
+                        delegate: MenuItem {
+                                    property bool downloaded: installed
+                                    text: title
+                                    Component.onCompleted: {
+                                        if (text && (text === searchEngineConfig.value)) {
+                                            searchEngine.currentIndex = index
+                                        }
                                     }
-                                }
-                            }
                         }
                     }
                 }
 
                 onCurrentItemChanged: {
                     if (currentItem.text !== searchEngineConfig.value) {
-                        searchEngineConfig.value = currentItem.text
+                        //TODO: downloaded is not updated until settings page is reopened
+                        if (currentItem.downloaded) {
+                            searchEngineConfig.value = currentItem.text
+                        } else {
+                            SearchEngineModel.install(currentItem.text)
+                        }
```

Comment:
I'd still move this whole onCurrentItem change log to the onClicked handler of the MenuItem component itself. There's no need to have this handler here as all the steps can be done already on onClicked inside the delegate.

Reply:
Yep, that move solved my problems.